### PR TITLE
AB#88163 Use data-testId instead of id

### PR DESCRIPTION
**Pull request by jjankowski87**

### CODE REVIEW

File: src/HE.Investment.AHP.WWW.Tests/Views/Application/NameTests.cs [hunk 1]
```diff
@@ -40,7 +40,7 @@ private static void AssertView(IHtmlDocument document, string? errorMessage = nu
             .HasElementWithText("p", "Each application must be for a single tenure. If you are developing a multi-tenure site, each tenure must be applied for within a separate application.")
             .HasElementWithText("p", "Each application needs a unique name. You will not be able to edit this later.")
             .HasElementWithText("p", "You should include the tenure type within your application name. For example, Village Way – Affordable Rent or Village Way – Shared Ownership.")
-            .HasContinueButton()
+            .HasGdsContinueButton()
```
Comment (Antek012) [inline src/HE.Investment.AHP.WWW.Tests/Views/Application/NameTests.cs:43]: Do we need "gds" in  name? 
All buttons should be gds and we can have abstraction for gds.

File: src/HE.Investments.TestsUtils/Extensions/HtmlFluentExtensions.cs [hunk 2]
```diff
@@ -66,41 +60,30 @@ public static IHtmlDocument HasValidationMessages(this IHtmlDocument htmlDocumen
...
-        var element = htmlDocument.GetElementById(id);
-
-        var anchorElement = element as IHtmlAnchorElement;
-        anchorElement.Should().NotBeNull($"Element with id {id} should be IHtmlAnchorElement");
-        htmlElement = anchorElement!;
-
-        return htmlDocument;
-    }
-
     public static IHtmlDocument HasLinkWithTestId(this IHtmlDocument htmlDocument, string testId, out IHtmlAnchorElement htmlElement)
```
Comment (Antek012) [inline src/HE.Investments.TestsUtils/Extensions/HtmlFluentExtensions.cs:63]: It this is final version maybe we can move link methods to HtmlDocumentLinkExtensions?

**Review by Antek012 — APPROVED** [review]

**Review by Carq — APPROVED** [review]

File: src/HE.Investment.AHP.WWW.Tests/Views/Application/NameTests.cs (hunk 1 above)
Comment (jjankowski87) [reply src/HE.Investment.AHP.WWW.Tests/Views/Application/NameTests.cs:43]: Removed Gds from text from method names

File: src/HE.Investments.TestsUtils/Extensions/HtmlFluentExtensions.cs (hunk 2 above)
Comment (jjankowski87) [reply src/HE.Investments.TestsUtils/Extensions/HtmlFluentExtensions.cs:63]: moved to link extension methods

**Pull request merged**